File: app/src/main/java/com/sube/movil/MainActivity.java
package com.sube.movil;
import android.content.Context;
import android.content.Intent;
import android.location.LocationManager;
import android.net.Uri;
import android.os.Bundle;
import android.provider.Settings;
import android.support.annotation.NonNull;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentTransaction;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;

import com.afollestad.materialdialogs.DialogAction;
import com.afollestad.materialdialogs.MaterialDialog;
import com.crashlytics.android.Crashlytics;
import com.special.ResideMenu.ResideMenu;
import com.special.ResideMenu.ResideMenuItem;
import org.json.JSONArray;

import hotchemi.android.rate.AppRate;
import hotchemi.android.rate.OnClickButtonListener;
import io.fabric.sdk.android.Fabric;


public class MainActivity extends FragmentActivity implements View.OnClickListener , ListInterface{

    private static ResideMenu resideMenu;
    private static ResideMenuItem mapa;
    private ResideMenuItem misube;
    private ResideMenuItem centro_ventas;
    private ResideMenuItem saldo;
    private ResideMenuItem contacto;
    private ResideMenuItem compartir;
    fragment2 fragment2;
    LocationManager lm;


    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        AppRate.with(this)//Llamo ventana para calificar aplicacion
                .setInstallDays(0) // default 10, 0 means install day.
                .setLaunchTimes(3) // default 10
                .setRemindInterval(1) // default 1
                .setShowLaterButton(true) // default true
                .setDebug(false) // default false
                .setOnClickButtonListener(new OnClickButtonListener() { // callback listener.
                    @Override
                    public void onClickButton(int which) {
                        Log.d(MainActivity.class.getName(), Integer.toString(which));
                    }
                })
                .monitor();

        // Show a dialog if meets conditions
        AppRate.showRateDialogIfMeetsConditions(this);
        setUpMenu();
        Fabric.with(this, new Crashlytics());
        fragment2 = new fragment2();
        if( savedInstanceState == null )
            changeFragment(new fragment1());
           checkLocation();
    }


    private void setUpMenu() {

        // attach to current activity;
        resideMenu = new ResideMenu(this);
        resideMenu.setBackground(R.drawable.fondomain);
        resideMenu.attachToActivity(this);
        resideMenu.setMenuListener(menuListener);
        resideMenu.setScaleValue(0.6f);

        // create menu items;
        mapa     = new ResideMenuItem(this, R.drawable.puntosrecarga, "Mapa");
        centro_ventas  = new ResideMenuItem(this, R.drawable.cuenta,  "Centros"+"\n"+"Venta"+"\n"+"Recarga");
        misube = new ResideMenuItem(this, R.drawable.horarios, "Mi Sube");
        contacto = new ResideMenuItem(this, R.drawable.contacto,"Contacto");
        compartir = new ResideMenuItem(this, R.drawable.compartir,"Compartí");
        saldo= new ResideMenuItem(this,R.drawable.horarios, "Mi saldo");


        mapa.setOnClickListener(this);
        centro_ventas.setOnClickListener(this);
        misube.setOnClickListener(this);
        saldo.setOnClickListener(this);
        contacto.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(Intent.ACTION_SENDTO);
                intent.setData(Uri.parse("mailto:")); // only email apps should handle this
                intent.putExtra(Intent.EXTRA_EMAIL, "marceloespinoza00@gmail.com");
                intent.putExtra(Intent.EXTRA_SUBJECT, "Sube Móvil");
                if (intent.resolveActivity(getPackageManager()) != null) {
                    startActivity(intent);
                }
            }
        });
        compartir.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent sendIntent = new Intent();
                sendIntent.setAction(Intent.ACTION_SEND);
                sendIntent.putExtra(Intent.EXTRA_TEXT, "https://play.google.com/store/apps/details?id=com.sube.movil");
                sendIntent.setType("text/plain");
                startActivity(sendIntent);
            }
        });
        resideMenu.setSwipeDirectionDisable(ResideMenu.DIRECTION_RIGHT);
        resideMenu.setSwipeDirectionDisable(ResideMenu.DIRECTION_LEFT);
        resideMenu.addMenuItem(misube, ResideMenu.DIRECTION_LEFT);
        resideMenu.addMenuItem(mapa, ResideMenu.DIRECTION_LEFT);
        resideMenu.addMenuItem(centro_ventas, ResideMenu.DIRECTION_RIGHT);
        resideMenu.addMenuItem(contacto, ResideMenu.DIRECTION_LEFT);
        resideMenu.addMenuItem(compartir,ResideMenu.DIRECTION_RIGHT);
        resideMenu.addMenuItem(saldo,ResideMenu.DIRECTION_RIGHT);
        // You can disable a direction by setting ->
        // resideMenu.setSwipeDirectionDisable(ResideMenu.DIRECTION_RIGHT);

        findViewById(R.id.title_bar_left_menu).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                resideMenu.openMenu(ResideMenu.DIRECTION_LEFT);
            }
        });
        findViewById(R.id.title_bar_right_menu).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                resideMenu.openMenu(ResideMenu.DIRECTION_RIGHT);
            }
        });
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent ev) {
        return resideMenu.dispatchTouchEvent(ev);
    }

    @Override
    public void onClick(View view) {

        if (view == mapa){
            changeFragment(new fragment1());
        }else if (view == centro_ventas){
            changeFragment(new fragment2());
        }else if (view == saldo) {
            changeFragment(new fragment3());
        }else if (view == misube) {
            changeFragment(new fragment4());}
        resideMenu.closeMenu();
    }

    private ResideMenu.OnMenuListener menuListener = new ResideMenu.OnMenuListener() {
        //llama cuando el menu esta abierto
        @Override
        public void openMenu() {

        }

        //llama cuando el menu esta cerrado
        @Override
        public void closeMenu() {

        }
    };

    private void changeFragment(Fragment targetFragment){
        if(targetFragment instanceof fragment2){
            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.main_fragment, fragment2, "fragment2")
                    .setTransitionStyle(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                    .commit();
        }else{

            getSupportFragmentManager().beginTransaction()
                    .replace(R.id.main_fragment, targetFragment, "fragment")
                    .setTransitionStyle(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                    .commit();
        }

    }

    public void checkLocation(){
        Boolean network_enabled=null;
        lm =(LocationManager) getApplicationContext().getSystemService(Context.LOCATION_SERVICE);
        network_enabled = lm.isProviderEnabled(LocationManager.NETWORK_PROVIDER);
        if(!network_enabled){
            new MaterialDialog.Builder(this).title("Advertencia").content("Para visualizar los puntos de venta debe permitir el acceso a su ubicacion mediante Wifi y Redes Moviles")
                    .negativeText("Cancelar").neutralText("Activar").
                    onNeutral(new MaterialDialog.SingleButtonCallback() {
                        @Override
                        public void onClick(@NonNull MaterialDialog dialog, @NonNull DialogAction which) {
                            Intent viewIntent = new Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS);
                            startActivity(viewIntent);
                        }
                    }).show();
        }

    }

    // What good method is to access resideMenu？
    public ResideMenu getResideMenu(){

        return resideMenu;
    }

    @Override
    public void onCreateList(JSONArray list) {

    }
}

File: app/src/main/java/com/sube/movil/Views/MainActivity.kt
package com.sube.movil.Views

import android.content.Context
import android.content.Intent
import android.location.LocationManager
import android.net.Uri
import android.os.Bundle
import android.provider.Settings
import android.util.Log
import android.view.MotionEvent
import android.view.View
import androidx.core.splashscreen.SplashScreen.Companion.installSplashScreen
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import androidx.fragment.app.FragmentTransaction
import com.afollestad.materialdialogs.MaterialDialog
import com.special.ResideMenu.ResideMenu
import com.special.ResideMenu.ResideMenuItem
import com.sube.movil.ListInterface
import com.sube.movil.R
import hotchemi.android.rate.AppRate
import org.json.JSONArray

class MainActivity : FragmentActivity(), View.OnClickListener, ListInterface {
    private var misube: ResideMenuItem? = null
    private var saldo: ResideMenuItem? = null
    private var fragment2: fragment2? = null
    private var lm: LocationManager? = null
    lateinit var context: Context

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        installSplashScreen()
        setContentView(R.layout.activity_main)
        context = applicationContext
        AppRate.with(this) //Llamo ventana para calificar aplicacion
            .setInstallDays(2) // default 10, 0 means install day.
            .setLaunchTimes(9) // default 10
            .setRemindInterval(1) // default 1
            .setShowLaterButton(true) // default true
            .setDebug(false) // default false
            .setOnClickButtonListener { which ->
                // callback listener.
                Log.d(MainActivity::class.java.name, which.toString())
            }
            .monitor()

        // Show a dialog if meets conditions
        AppRate.showRateDialogIfMeetsConditions(this)
        setUpMenu()
        fragment2 = fragment2()
        if (savedInstanceState == null) changeFragment(MapFragment())
        checkLocation()
    }

    private fun setUpMenu() {

        // attach to current activity;
        Companion.resideMenu = ResideMenu(this)
        Companion.resideMenu!!.setBackground(R.drawable.fondomain)
        Companion.resideMenu!!.attachToActivity(this)
        Companion.resideMenu!!.menuListener = menuListener
        Companion.resideMenu!!.setScaleValue(0.6f)

        // create menu items;
        mapa = ResideMenuItem(this, R.drawable.puntosrecarga, "Mapa")
        misube = ResideMenuItem(this, R.drawable.horarios, "Mi Sube")
        val contacto = ResideMenuItem(this, R.drawable.contacto, "Contacto")
        val compartir = ResideMenuItem(this, R.drawable.compartir, "Compartí")
        saldo = ResideMenuItem(this, R.drawable.horarios, "Mi saldo")
        mapa!!.setOnClickListener(this)
        misube!!.setOnClickListener(this)
        saldo!!.setOnClickListener(this)
        contacto.setOnClickListener(View.OnClickListener {
            val intent = Intent(Intent.ACTION_SENDTO)
            intent.data = Uri.parse("mailto:") // only email apps should handle this
            intent.putExtra(Intent.EXTRA_EMAIL, "marceloespinoza00@gmail.com")
            intent.putExtra(Intent.EXTRA_SUBJECT, "Sube Móvil")
            if (intent.resolveActivity(packageManager) != null) {
                startActivity(intent)
            }
        })
        compartir.setOnClickListener(View.OnClickListener {
            val sendIntent = Intent()
            sendIntent.action = Intent.ACTION_SEND
            sendIntent.putExtra(
                Intent.EXTRA_TEXT,
                "https://play.google.com/store/apps/details?id=com.sube.movil"
            )
            sendIntent.type = "text/plain"
            startActivity(sendIntent)
        })
        Companion.resideMenu!!.setSwipeDirectionDisable(ResideMenu.DIRECTION_RIGHT)
        Companion.resideMenu!!.setSwipeDirectionDisable(ResideMenu.DIRECTION_LEFT)
        Companion.resideMenu!!.addMenuItem(misube, ResideMenu.DIRECTION_LEFT)
        Companion.resideMenu!!.addMenuItem(mapa, ResideMenu.DIRECTION_LEFT)
        Companion.resideMenu!!.addMenuItem(contacto, ResideMenu.DIRECTION_LEFT)
        Companion.resideMenu!!.addMenuItem(compartir, ResideMenu.DIRECTION_RIGHT)
        Companion.resideMenu!!.addMenuItem(saldo, ResideMenu.DIRECTION_RIGHT)
        // You can disable a direction by setting ->
        // resideMenu.setSwipeDirectionDisable(ResideMenu.DIRECTION_RIGHT);
        findViewById<View>(R.id.title_bar_left_menu).setOnClickListener(
            View.OnClickListener { Companion.resideMenu!!.openMenu(ResideMenu.DIRECTION_LEFT) })
        findViewById<View>(R.id.title_bar_right_menu).setOnClickListener(
            View.OnClickListener { Companion.resideMenu!!.openMenu(ResideMenu.DIRECTION_RIGHT) })
    }

    override fun dispatchTouchEvent(ev: MotionEvent): Boolean {
        return Companion.resideMenu!!.dispatchTouchEvent(ev)
    }

    override fun onClick(view: View) {
        if (view === mapa) {
            changeFragment(MapFragment())
        } else if (view === saldo) {
            changeFragment(SaldoFragment())
        } else if (view === misube) {
            changeFragment(fragment4())
        }
        Companion.resideMenu!!.closeMenu()
    }

    private val menuListener: ResideMenu.OnMenuListener = object : ResideMenu.OnMenuListener {
        //llama cuando el menu esta abierto
        override fun openMenu() {}

        //llama cuando el menu esta cerrado
        override fun closeMenu() {}
    }

    private fun changeFragment(targetFragment: Fragment) {
        if (targetFragment is fragment2) {
            fragment2?.let {
                supportFragmentManager.beginTransaction()
                    .replace(R.id.main_fragment, it, "fragment2")
                    .setTransitionStyle(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                    .commit()
            }
        } else {
            supportFragmentManager.beginTransaction()
                .replace(R.id.main_fragment, targetFragment, "fragment")
                .setTransitionStyle(FragmentTransaction.TRANSIT_FRAGMENT_FADE)
                .commit()
        }
    }

    private fun checkLocation() {
        var network_enabled: Boolean? = null
        lm = applicationContext.getSystemService(Context.LOCATION_SERVICE) as LocationManager
        network_enabled = lm!!.isProviderEnabled(LocationManager.NETWORK_PROVIDER)
        if (!network_enabled) {
           /* Builder(this).title("Advertencia")
                .content("Para visualizar los puntos de venta debe permitir el acceso a su ubicacion mediante Wifi y Redes Moviles")
                .negativeText("Cancelar")
                .neutralText("Activar")
                .onNeutral(object : SingleButtonCallback() {
                    fun onClick(dialog: MaterialDialog, which: DialogAction) {
                        val viewIntent = Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS)
                        startActivity(viewIntent)
                    }
                }).show()*/
            MaterialDialog(this).show {
                title(text = "Advertencia")
                message (text = "Para visualizar los puntos de venta debe permitir el acceso a su ubicacion mediante Wifi y Redes Moviles")

                positiveButton(text = "Aceptar") { dialog ->
                            val viewIntent = Intent(Settings.ACTION_LOCATION_SOURCE_SETTINGS)
                            startActivity(viewIntent)
                }

                negativeButton(text = "Cancelar") { dialog ->
                    dialog.dismiss()
                }
            }
        }
    }

    // What good method is to access resideMenu？
    val resideMenu: ResideMenu?
        get() = Companion.resideMenu

    override fun onCreateList(list: JSONArray) {}

    companion object {
        private var resideMenu: ResideMenu? = null
        private var mapa: ResideMenuItem? = null
        var context: Context? = null
            private set
    }
}